src/EditorFeatures/Core/Implementation/SmartIndent/SmartIndent.cs

```diff
@@ -40,20 +44,38 @@ public void Dispose()
             using (Logger.LogBlock(FunctionId.SmartIndentation_Start, cancellationToken))
             {
                 var document = lineToBeIndented.Snapshot.GetOpenDocumentInCurrentContextWithChanges();
-                var syncService = document?.GetLanguageService<ISynchronousIndentationService>();
+                if (document == null)
+                {
+                    return null;
+                }
+
+                // First, try to go through the normal feature-layer indentation service.
+                var newService = document.GetLanguageService<NewIndentationService>();
+                if (newService != null)
+                {
+                    var result = newService.GetDesiredIndentation(document, lineToBeIndented.LineNumber, cancellationToken);
+                    return result.GetIndentation(_textView, lineToBeIndented);
+                }
 
-                if (syncService != null)
+                // If we don't have a feature-layer service, try to fall back to the legacy
+                // editor-feature-layer interfaces.
+
+#pragma warning disable CS0618 // Type or member is obsolete
+                var oldSyncService = document.GetLanguageService<OldSynchronousIndentationService>();
```

Comment:
can we not do this? why not just delete these?

Reply:
so, I just saw up there that indentaiton now can never be null, so I am not sure when this can ever return null? can we add assert that confirm we never return null now.